src/wrappers/ghs_wrapper.cpp

```diff
@@ -133,4 +133,9 @@ std::string ghs_wrapper_t::get_program_id() {
 
   return HASH_VERSION + program_version_info + os_version_info;
 }
+
+string_list_t ghs_wrapper_t::get_capabilities() {
+  // GHS compilers implicitly create target directories for object files:
+  return gcc_wrapper_t::get_capabilities() += "create_target_dirs";
```

Comment:
Rather than using `+=` in a return-statement, I prefer the more verbose:

```c++
  auto caps = gcc_wrapper_t::get_capabilities();
  caps += "create_target_dirs";
  return caps;
```

Reply:
...or add `operator+()` to `string_list_t`:

```c++
  string_list_t operator+(const std::string& str) {
    string_list_t result(*this);
    result += str;
    return result;
  }

  string_list_t operator+(const string_list_t& list) {
    string_list_t result(*this);
    result += list;
    return result;
  }
```